### PR TITLE
New geturl param to force IPv4 or IPv6

### PR DESCRIPTION
**Pull request by rhansen**
This will be used by the upcoming `webv4` and `webv6` options to ensure that the checkip service returns the desired type of IP address.

Addresses #172 

cc @dkerr64 @abelbeck

### CODE REVIEW

File: ddclient [hunk 1]
```diff
@@ -1988,128 +1993,133 @@ EOM
...
 ######################################################################
 sub geturl {
-    my $proxy    = shift || '';
-    my $url      = shift || '';
-    my $login    = shift || '';
-    my $password = shift || '';
-    my $headers  = shift || '';
-    my $method   = shift || 'GET';
-    my $data     = shift || '';
+    my ($p) = @_;
```
Comment (SuperSandro2000) [inline ddclient:1996]: I am not a big fan of the single p variable. Maybe we find a good three letter one?

Comment (rhansen) [reply ddclient]: You haven't written much Go code, have you? :stuck_out_tongue_winking_eye: 

I decided to keep the original variables.

File: ddclient [hunk 2]
```diff
@@ -2068,56 +2075,59 @@ sub geturl {
...
-            MultiHomed => 1,
-            Timeout    => opt('timeout'),
-        );
-        defined $sd or warning("cannot connect to %s:%s socket: %s %s", $peer, $port, $@, IO::Socket::SSL::errstr());
-    } elsif ($globals{'ipv6'}) {
+    my $socket_class = IO::Socket::INET;
+    if ($use_ssl) {
+        # IO::Socket::SSL will load IPv6 support if available on the system.
+        load_ssl_support;
+        $socket_class = IO:Socket::SSL;
```
Comment (SuperSandro2000) [inline ddclient:2082]:
```suggestion
        $socket_class = IO::Socket::SSL;
```

Comment (rhansen) [reply ddclient]: Whoops, thank you.

**Comment by SuperSandro2000** [comment]
Also you did not test this very well:

```
Bareword "IO::Socket::INET" not allowed while "strict subs" in use at ./ddclient line 2078.
Bareword "IO::Socket::SSL" not allowed while "strict subs" in use at ./ddclient line 2082.
Bareword "IO::Socket::INET6" not allowed while "strict subs" in use at ./ddclient line 2085.
Execution of ./ddclient aborted due to compilation errors.
```

**Comment by rhansen** [comment]
> Also you did not test this very well:

No, I didn't. If we think that PR #161 will be merged soon I'll write some unit tests for this.

**Comment by SuperSandro2000** [comment]
You should test it anyway cause right now this breaks ddclient and I can't merge it.

**Comment by rhansen** [comment]
I did run some manual tests with all combinations of with/without SSL and unforced/force-v4/force-v6 and they all worked.

I'm also writing unit tests so that we can be comfortably confident. I'm going to change this to draft until the tests have been written.

**Comment by dkerr64** [comment]
I've also tested with forced/unforced and it works.  It is notable that you do get different results between unforced and the forced for the web method.  Some websites that report your IP do it based on whether you connect over IPv4 or IPv6.  With the unforced method you have no control over this... it is left up to the underlying network stack in the operating system and it could decide to use IPv4 or IPv6 on a whim (well, actually I think the rules say to bias towards IPv6 unless IPv4 is going to be faster).  But it does mean that the results of a web method to the same site could change from time-to-time.

The bottom line being that the quicker we can depreciate the old settings the better.

David

**Comment by rhansen** [comment]
Testing this has turned out to be more obnoxious than I had hoped, and I don't want to hold up this PR while I work out the kinks. I've manually tested all of the cases and believe the behavior to be correct.
